pycqed/analysis_v2/residual_ZZ_analysis.py

```diff
@@ -0,0 +1,192 @@
+                frequency = fit_res.best_values['frequency']
+                if y[1]-y[0] < 0:
+                    frequency = -1*frequency
+                if np.std(y)/np.mean(y) < 0.1:
+                    frequency = 0
+                residual_ZZ_matrix[self.qubits.index(target_qubit), self.qubits.index(active_spectator)] = frequency
+                print('{}:Setting {},{} to {}'.format(ts, target_qubit, active_spectator, frequency))
+            for i in range(len(self.qubits)):
+                residual_ZZ_matrix[i,i] = np.nan
+            self.proc_data_dict['quantities_of_interest'] = {
+                'matrix': residual_ZZ_matrix,
+                'qubit_labels': self.qubits}
+    def prepare_plots(self):
+        self.plot_dicts['residual_matrix'] = {
+            'plotfn': plot_crosstalk_matrix,
+            'crosstalk_matrix': self.proc_data_dict['quantities_of_interest']['matrix'],
+            'qubit_labels': self.qubits
+        }
+
+def fit_residual_coupling_oscillation(x, y, **kw):
+    average = np.mean(y)
+    ft_of_data = np.fft.fft(y)
+    index_of_fourier_maximum = np.argmax(np.abs(
+    ft_of_data[1:len(ft_of_data) // 2])) + 1
+    max_ramsey_delay = x[-1] - x[0]
+    fft_axis_scaling = 1 / (max_ramsey_delay)
+    freq_est = fft_axis_scaling * index_of_fourier_maximum
+    est_number_of_periods = index_of_fourier_maximum
+    
+    
+    damped_osc_mod = lmfit.Model(fit_mods.ExpDampOscFunc)
+    damped_osc_mod.set_param_hint('frequency',
+                                  value=freq_est,
+                                  vary=True)
+    damped_osc_mod.set_param_hint('phase',
+                                  value=np.pi/2,vary=False)
+    damped_osc_mod.set_param_hint('amplitude',
+                                  value=0.5*(max(y)-min(y)),
+                                  min=min(y),
+                                  max=max(y),
+                                  vary=True)
+    damped_osc_mod.set_param_hint('tau',
+                                  value=x[1]*10,
+#                                   min=x[1],
+#                                   max=x[1]*1000,
+                                  vary=True)
+    damped_osc_mod.set_param_hint('exponential_offset',
+                                  value=average,
+                                  min=min(y),
+                                  max=max(y),
+                                  vary=True)
+    damped_osc_mod.set_param_hint('oscillation_offset',
+                                  value=0,
+                                  vary=False)
+    damped_osc_mod.set_param_hint('n',
+                                  value=1,
+                                  vary=False)
+    params = damped_osc_mod.make_params()
+
+    fit_res = damped_osc_mod.fit(data=y,
+                                 t=x,
+                                 params=params)
+    return fit_res
+
+def plot_crosstalk_matrix(crosstalk_matrix,
```

Comment:
This function could use a docstring. 
What is the crosstalk matrix? 
What are the input arguments? 
Are there assumptions on the shape of the crosstalk matrix (e.g., has to be n*n) and the units? 
Same for qubit_labels, what are "combinations"? 
Is there a way to specify the limits to the cbar?